src/geode/model/helpers/convert_to_mesh.cpp

```diff
@@ -539,6 +553,319 @@ namespace
+                            unique_vertex );
+                    }
+                    else
+                    {
+                        const auto new_vertex_index = mesh_builder.create_point(
+                            surface_mesh.point( vertex ) );
+                        polygon[v] = new_vertex_index;
+                        model2mesh.unique_vertices_mapping.map(
+                            unique_vertex, new_vertex_index );
+                    }
+                }
+                polygons[p] = mesh_builder.create_polygon( polygon );
+                model2mesh.surface_polygons_mapping.map(
+                    { surface.id(), p }, polygons[p] );
+            }
+            compute_polygons_surface_adjacencies< Model::dim >(
+                polygons, surface_mesh, mesh_builder );
+        }
+    }
+
+    template < typename Model, typename MeshType >
+    void map_line_edges(
+        Model& model, geode::ModelToMeshMappings& model2mesh, MeshType& mesh )
+    {
+        mesh.enable_edges();
+        for( const auto& line : model.lines() )
+        {
+            const auto& line_mesh = line.mesh();
+            for( const auto line_edge : geode::Range{ line_mesh.nb_edges() } )
+            {
+                auto unique_vertices =
+                    geode::edge_unique_vertices( model, line, line_edge );
+                for( const auto edge_vertex_id :
+                    geode::LRange{ unique_vertices.size() } )
+                {
+                    unique_vertices[edge_vertex_id] =
+                        model2mesh.unique_vertices_mapping.in2out(
+                            unique_vertices[edge_vertex_id] );
+                }
+                const auto solid_edge =
+                    mesh.edges().edge_from_vertices( unique_vertices ).value();
+                model2mesh.line_edges_mapping.map(
+                    { line.id(), line_edge }, solid_edge );
+            }
+        }
+    }
+
+    template < typename Model >
+    std::tuple< std::unique_ptr< geode::SurfaceMesh< Model::dim > >,
+        geode::ModelToMeshMappings >
+        new_convert_model_into_surface( const Model& model )
+    {
+        std::vector<
+            std::reference_wrapper< const geode::SurfaceMesh< Model::dim > > >
+            meshes;
+        meshes.reserve( model.nb_surfaces() );
+        for( const auto& surface : model.surfaces() )
+        {
+            meshes.emplace_back( surface.mesh() );
+        }
+        auto mesh =
+            geode::detail::create_mesh< geode::SurfaceMesh< Model::dim > >(
+                meshes );
+        auto mesh_builder =
+            geode::SurfaceMeshBuilder< Model::dim >::create( *mesh );
+        geode::ModelToMeshMappings model2mesh;
+        build_polygons_from_model( model, *mesh_builder, model2mesh );
+        map_line_edges( model, model2mesh, *mesh );
+        map_corner_vertices( model, model2mesh );
+        mesh_builder->compute_polygon_adjacencies();
```

Comment:
I'd compute the adjacencies before adding the mappings, in case it speeds up the enable_edges process, but it should have the same results